assignment/JS_DOM/script.js

```diff
@@ -0,0 +1,119 @@
+(function (scope) {
+    "use strict";
+
+    var form = document.querySelector('#new-todo');
+    var editForm = document.querySelector('#edit');
+    var tasksContainer = document.querySelector('#tasks');
+    var taskManager = createTaskManager();
+    taskManager.onChange(update);
+    loadTasks();
+
+    form && form.addEventListener('submit', addTask);
+    
+    function addTask(event) {
+        event.preventDefault();
+        var task = {};
+        event.target.querySelectorAll('input:not([type="submit"]').forEach(function (input) {
+            task[input.name] = input.value;
+            input.value = null;
+        });
+        taskManager.create(task.category, task.title, task.priority, task.estimate);
+    }
+
+    function update(tasks) {
+        localStorage.setItem('tasks',JSON.stringify(tasks));
+        while (tasksContainer.hasChildNodes()) {
+            tasksContainer.removeChild(tasksContainer.lastChild);
+        }
+        taskManager.getAll().forEach(function (task) {
+            tasksContainer.appendChild(createTaskRow(task));
+        });
+    }
+
+    function createTaskRow(task) {
+        var tr = document.createElement('tr');
+        
+        var deleteLinkTd = document.createElement('td');
+        var link = document.createElement('a');
+        link.innerHTML = 'Delete';
+        deleteLinkTd.appendChild(link);
+
+        var editLinkTd = document.createElement('td');
+        var editLink = document.createElement('a');
+        editLink.innerHTML = 'Edit';
+        editLinkTd.appendChild(editLink);
+
+        link.addEventListener("click",function(){
+        console.log("remove it");
+        taskManager.remove(task);
+           });
+
+        editLink.addEventListener("click",function(){
+            getTaskData(task);
+        });
+        function editTask(event){
+            event.preventDefault();
+            task.category = document.getElementById('edit-category').value;
```

Comment:
You could have used a loop to extract the fields values like we did in `addTask()`
```
 event.target.querySelectorAll('input:not([type="submit"]').forEach(function (input) {
             task[input.name] = input.value;
             input.value = null;
         });
```